tests/functional/conftest.py

```diff
@@ -210,3 +212,42 @@ def dependency_config(temp_config):
 @pytest.fixture
 def base_projects_directory():
     return BASE_PROJECTS_DIRECTORY
+
+
+@pytest.fixture
+def chain_at_block_5(chain):
+    snapshot_id = chain.snapshot()
+    chain.mine(5)
+    yield chain
+    chain.restore(snapshot_id)
+
+
+class PollDaemon(threading.Thread):
+    def __init__(self, name, poller, handler, stop_condition, *args, **kwargs):
+        super().__init__(*args, name=f"ape_poll_{name}", **kwargs)
+        self._poller = poller
+        self._handler = handler
+        self._do_stop = stop_condition
+
+    def __enter__(self):
+        self.start()
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.stop()
+
+    def run(self):
+        while True:
+            if self._do_stop():
+                return
+
+            # WARN: Will wait indefinitely for more events
```

Comment:
Is there a way to set a timeout in `__init__` and respect it here?

Reply:
This was a good suggestion. I just added a `new_block_timeout` kwarg to both polling methods so that if a new block has not appeared in X number of seconds (float), raise an exception.

* The default for local in 10 seconds, just so tests don't hang on accident
* And the default for live networks is 3 * the `block_time` network setting because that seems okay for now